modules/augment/src/rotate.cpp

```diff
@@ -0,0 +1,40 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+namespace cv {
+namespace augment {
+
+    Rotate::Rotate(Range _angleRange)
+    {
+        this->angleRange = _angleRange;
+        this->resetRandom();
+    }
+
+    Rotate::Rotate() : Rotate::Rotate(Range(0, 180)) {}
+
+    void Rotate::resetRandom()
+    {
+        this->currentAngle = rng.uniform(this->angleRange.start, this->angleRange.end + 1);
+    }
+
+    void Rotate::image(InputArray _src, OutputArray _dst)
+    {
+        Mat src = _src.getMat();
+        Mat rotationMat = getRotationMatrix2D(Point2f(src.cols / 2, src.rows / 2), this->currentAngle, 1);
+        warpAffine(src, _dst, rotationMat, Size(src.cols, src.rows));
```

Comment:
As `warpAffine` also has such parameters as interpolation flag and border mode, it'd be nice to have an opportunity to have an ability to pick them randomly as well. But let's do it in the future PRs.

Reply:
Okay I will look into that when we resolve the rest of the issues